### PR TITLE
Add 2pc transaction approach to InsertOrder

### PR DESCRIPTION
**Pull request by Sugar-pack**

### CODE REVIEW

File: internal/grpcapi/order_service.go [hunk 1]
```diff
@@ -2,29 +2,59 @@ package grpcapi
...
 	"context"
+	"database/sql"
+	"fmt"
 
-	pb2 "github.com/Sugar-pack/orders-manager/pkg/pb"
 	"github.com/Sugar-pack/users-manager/pkg/logging"
 	"github.com/google/uuid"
 	"github.com/jmoiron/sqlx"
+
+	pb2 "github.com/Sugar-pack/orders-manager/pkg/pb"
```
Comment (pzabolotniy) [inline internal/grpcapi/order_service.go:12]: is alias **pb2** required?

File: internal/db/prepare_transaction.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+type Order struct {
+	ID        uuid.UUID `db:"id"`
+	UserID    uuid.UUID `db:"user_id"`
+	Label     string    `db:"label"`
+	CreatedAt time.Time `db:"created_at"`
+}
+
+func InsertUser(ctx context.Context, dbConn sqlx.ExecerContext, order *Order) error {
+	query := "INSERT INTO orders ( id,  user_id, label, created_at ) VALUES ($1, $2, $3, $4)"
```
Comment (pzabolotniy) [inline internal/db/prepare_transaction.go:20]: try to use named placeholders, not positional.
named parameters are more maintainable.

File: internal/grpcapi/order_service.go [hunk 3]
```diff
@@ -2,32 +2,75 @@ package grpcapi
...
 	logger.Info("ReceiveOrder")
 	orderID := uuid.New()
 	txID := uuid.New()
-	query := "INSERT INTO orders ( id,  user_id, label, created_at ) VALUES ($1, $2, $3, $4)"
-	_, err := s.dbConn.Exec(query, orderID.String(), order.UserId, order.Label, order.CreatedAt.AsTime())
+	parseUserID, err := uuid.Parse(order.UserId)
+	if err != nil {
+		logger.Error("Error parsing user id ", err)
+
+		return nil, fmt.Errorf("error parsing user id %w", err)
```
Comment (pzabolotniy) [inline internal/grpcapi/order_service.go:31]: grpc error should contain error-code.
take a look https://jbrandhorst.com/post/grpc-errors/

Comment (Sugar-pack) [reply internal/grpcapi/order_service.go]: done

**Pull request merged**